### DISCUSSION TITLE
Small interpreter

### DISCUSSION THREAD

**Discussion by andrewtholt** [arendst/Tasmota#13109] [Q&A] [answered]
Since some the boards (esp32_16M etc) now significant amount of memory has anybody considered a small interpreter to facilitate writing rules and/or action ? I'm thinking something simple such Lua or (for the more esoteric)  forth.

Any thoughts ?

**Comment by s-hadinger** [comment, marked as answer]
Your wishes are a reality. 
https://tasmota.github.io/docs/Berry/

And it's included by default in all esp32 firmwares

**Comment by andrewtholt** [comment]
Thanks, I'll take a look.
Andrew

**Comment by andrewtholt** [comment]
I suppose nothing similar for esp8266 based devices ?

**Reply by s-hadinger** [reply]
No, it's too limited in resources to be usable

**Reply by sfromis** [reply]
On ESP8266 Tasmota, the only alternative to rules is the https://tasmota.github.io/docs/Scripting-Language/
Takes some learning curve to grok, as it is not that similar to classic programming languages.

**Comment by andrewtholt** [comment]
Thanks, I'll check it out.
